codegen/gateway.go

```diff
@@ -373,6 +381,8 @@ func NewEndpointSpec(
 			)
 		}
 		workflowImportPath = iworkflowImportPath.(string)
+	} else if workflowType == "self-serving" {
```

Comment:
just a refactoring suggestion -- once we are in this realm of more than 2 string name comparisons, it may make sense to have them as "const" value